src/main/Server.java

```diff
@@ -0,0 +1,31 @@
+import clientHandler.ClientHandlerThread;
+import clientInteractors.ClientInteractorInterface;
+import clientInteractors.TCPClientInteractor;
+
+import java.io.IOException;
+
+public class Server {
+
+    public static void main(String[] args) {
+	    runServer();
+    }
+
+    private static void runServer() {
+        while(true){
+            ClientInteractorInterface clientInteractor = setUpInteraction();
+            new Thread(new ClientHandlerThread(clientInteractor)).start();
```

Comment:
The setUpInteraction() method waits (so it is blocking) for a connection, and then a new thread is created for each client which connects, so the server can continue listening for incoming connections.